libverify/verify_klee.c

```diff
@@ -0,0 +1,361 @@
+/**
+ * Defines assert, require and nd implementations for symbolic execution.
+ *  @date 2019
+ */
+
+#include "verify.h"
+
+#include "klee/klee.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <assert.h>
+
+void sol_setup(int _argc, const char **_argv)
+{
+}
+
+void sol_on_transaction(void){
+
+}
+
+void ll_assume(sol_raw_uint8_t _cond){
+	klee_assume(_cond);
+}
+
+void sol_assert(sol_raw_uint8_t _cond, const char* _msg)
+{
+    klee_assert(_cond);
+}
+
+void sol_require(sol_raw_uint8_t _cond, const char* _msg)
+{
+    klee_assume(_cond);
+}
+
+uint8_t rt_nd_byte(const char* _msg){
+    uint8_t res;
+    klee_make_symbolic(&res, sizeof(res), "rt_nd_byte");
```

Comment:
My one suggestion would be passing `_msg` to `klee_make_symbolic`. This way klee can give the variables more meaningful names.

`klee_make_symbolic(&res, sizeof(res), _msg);`